### DISCUSSION TITLE
Idea: allow RPC calls without specifying account number

### DISCUSSION THREAD

**Discussion by cuu508** [AsamK/signal-cli#1132] [Ideas]
If there is only one account configured, signal-cli can send messages without specifying the sender's number:

```
signal-cli send -m "Hi" +xxxxxxxxxx
```

But JSON-RPC needs an explicit account number. If I run signal-cli without specifying "-a":

```
./signal-cli daemon --socket
```

And try to send a message via JSON-RPC, I get the following message back:

```
{'jsonrpc': '2.0', 'error': {'code': -32602, 'message': 'Method requires valid account parameter', 'data': None}, 'id': '13a76726-93b9-44a8-9755-d6b625c32896'}
```

Perhaps JSON-RPC could also pick up the sender's number automatically, if there's only one?

Just an idea – there may be cons to this that I'm not aware of!

**Comment by AsamK** [comment]
Good point, I don't think there's a disadvantage. Other than possible confusion by the user, when a second account is registered ...
I've implemented it: 210466e7d9fe589ddfaba6d8ac82948bbea500d5
